Sources/swift-doc/Supporting Types/Components/Relationships.swift

```diff
@@ -73,30 +73,36 @@ struct Relationships: Component {
     }
 
     var html: HypertextLiteral.HTML {
-        var graph = symbol.graph(in: module)
-        guard !graph.edges.isEmpty else { return "" }
-
-        graph.aspectRatio = 0.125
-        graph.center = true
-        graph.overlap = "compress"
+        guard sections.map(\.symbols.count).reduce(0, +) != 0 else { return "" }
 
-        let algorithm: LayoutAlgorithm = graph.nodes.count > 3 ? .neato : .dot
         var svg: HypertextLiteral.HTML?
+        var graph = symbol.graph(in: module)
+        if !graph.edges.isEmpty {
+            graph.aspectRatio = 0.125
+            graph.center = true
+            graph.overlap = "compress"
+
+            let algorithm: LayoutAlgorithm = graph.nodes.count > 3 ? .neato : .dot
 
-        do {
-            svg = try HypertextLiteral.HTML(String(data: graph.render(using: algorithm, to: .svg), encoding: .utf8) ?? "")
-        } catch {
-            logger.error("\(error)")
+            do {
+                svg = try HypertextLiteral.HTML(String(data: graph.render(using: algorithm, to: .svg), encoding: .utf8) ?? "")
+            } catch {
+                logger.error("\(error)")
+            }
         }
 
         return #"""
         <section id="relationships">
             <h2 hidden>Relationships</h2>
-            <figure>
-                \#(svg ?? "")
+                \#(svg.flatMap { svg in
```

Comment:
I'm not sure what `flatMap` is providing here as compared to the original, but that could just be me having trouble reading the diff.

Reply:
I moved the "<figure>" block inside `flatMap` so it is not rendered when the `svg` data is missing.

Reply:
Ah, got it. Thanks for clarifying.